src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -225,6 +287,21 @@ bool QueueIsEmpty ()
 			return _methods.Count == 0;
 		}
 
+		bool ProcessCopyUsedAssemblies ()
```

Comment:
I do not understand what purpose this serves.  It doesn't look like enough to handle the copyused scenario.  Can you add tests to prove that copyused works as expected?  Without tests for copyused I don't see the value in adding code to handle it.  The behavior is obscure, there are zero tests today, and this bug has gone unnoticed for years.  Any code added to handle copyused could easily break and we wouldn't know for years again.

Reply:
Fair point, will add test for this path